zsh-history-substring-search.zsh

```diff
@@ -51,9 +52,10 @@ typeset -g _history_substring_search_refresh_display
 typeset -g _history_substring_search_query_highlight
 typeset -g _history_substring_search_result
 typeset -g _history_substring_search_query
-typeset -g -A _history_substring_search_raw_matches
```

Comment:
While I support this cleanup of the global variables, please leave it out of this PR, I will take care of it once merged.